File: sampleandroid/src/main/java/com/example/sampleandroid/view/model/ApiPollingScreen.kt
package com.example.sampleandroid.view.model

import androidx.compose.Composable
import androidx.ui.foundation.Text
import androidx.ui.layout.Column
import kim.jeonghyeon.androidlibrary.extension.resourceToString
import kim.jeonghyeon.sample.compose.R
import kim.jeonghyeon.sample.viewmodel.ApiPollingViewModel

class ApiPollingScreen(private val model: ApiPollingViewModel = ApiPollingViewModel()) : ModelScreen(model) {
    override val title: String = R.string.polling.resourceToString()

    @Composable
    override fun view() {
        Column {
            Text("fail count ${+model.count}")
            if (model.status.asValue().isSuccess()) {
                Text("result ${+model.result}")
            }
        }
    }

    @Composable
    override fun compose() {
        super.compose()
    }
}

File: sample/sample-native/kotlinIOS/view/model/ApiPollingScreen.swift
//
//  ApiPollingScreen.swift
//  kotlinIOS
//
//  Created by hyun kim on 19/07/20.
//  Copyright © 2020 hyun kim. All rights reserved.
//

import Foundation
import SwiftUI
import sample_base

struct ApiPollingScreen: SampleScreen {

    var model = ApiPollingViewModel()

    func content(navigator: Navigator) -> some View {
        VStack {
            Text("fail count \(model.count.value as! Int)")
            if (model.status.value!.isSuccess()) {
                Text("result \(model.result.value!)")
            }
        }
        .navigationTitle("Api Polling".localized())
    }

}
